torch/utils/data/dataloader.py

```diff
@@ -376,6 +376,12 @@ class DataLoader(object):
             worker subprocess with the worker id (an int in ``[0, num_workers - 1]``) as
             input, after seeding and before data loading. (default: None)
 
+    .. warning:: If :attr:`batch_sampler` is given, setting :attr:`batch_size`,
+                 :attr:`shuffle`, :attr:`sampler`, or :attr:`drop_last` is
+                 invalid because such information is provided by
+                 :attr:`batch_sampler`. In this case, the data loader will have
+                 ``dataloader.batch_size == dataloader.drop_last == dataloader.sampler == None``.
```

Comment:
Can we implement a check in `__setattr__`?

Reply:
You are right.